Classes/Repository/RepositoryClient.swift

```diff
@@ -0,0 +1,152 @@
+    }
+    
+    func nextPageToken(from data: GraphQLMappable) -> String? {
+        guard let issues = data as? RepoIssuesQuery.Data else { return nil }
+        guard let pageInfo = issues.repository?.issues.pageInfo, pageInfo.hasNextPage else { return nil }
+        return pageInfo.endCursor
+    }
+}
+
+extension RepoPullRequestsQuery: RepositoryQuery {
+    func summaryTypes(from data: GraphQLMappable) -> [IssueSummaryType] {
+        guard let prs = data as? RepoPullRequestsQuery.Data else { return [] }
+        return prs.repository?.pullRequests.nodes?.flatMap { $0 } ?? []
+    }
+    
+    func nextPageToken(from data: GraphQLMappable) -> String? {
+        guard let prs = data as? RepoPullRequestsQuery.Data else { return nil }
+        guard let pageInfo = prs.repository?.pullRequests.pageInfo, pageInfo.hasNextPage else { return nil }
+        return pageInfo.endCursor
+    }
+}
+
+final class RepositoryClient {
+
+    var issues = [IssueSummaryModel]()
+    var issuesNextPage: String?
+    
+    var pullRequests = [IssueSummaryModel]()
+    var pullRequestsNextPage: String?
+    
+    let githubClient: GithubClient
+    let repo: RepositoryLoadable
+    
+    init(githubClient: GithubClient, repo: RepositoryLoadable) {
+        self.githubClient = githubClient
+        self.repo = repo
+    }
+    
+    struct RepositoryLoadSuccessPayload {
+        let models: [IssueSummaryModel]?
+        let nextPage: String?
+    }
+    
+    enum RepoLoadResultType {
+        case error
+        case success(RepositoryLoadSuccessPayload)
+    }
+    
+    private func load<T: GraphQLQuery>(queryType: T.Type,
+                                       repo: RepositoryLoadable,
+                                       after: String? = nil,
+                                       containerWidth: CGFloat,
+                                       completion: @escaping (RepoLoadResultType) -> ()) where T: RepositoryQuery {
+        let query = queryType.init(owner: repo.owner, name: repo.name, after: after)
+        
+        githubClient.apollo.fetch(query: query, cachePolicy: .fetchIgnoringCacheData) { (result, error) in
+            guard error == nil, result?.errors == nil, let data = result?.data else {
+                ShowErrorStatusBar(graphQLErrors: result?.errors, networkError: error)
+                completion(.error)
+                return
+            }
+            
+            let summaries: [IssueSummaryModel] = query.summaryTypes(from: data).map { summaryType in
+                summaryType.attributedTitle.computeSize(containerWidth)
+                return IssueSummaryModel(info: summaryType)
+            }
+            
+            completion(.success(RepositoryLoadSuccessPayload(models: summaries, nextPage: query.nextPageToken(from: data))))
+        }
+    }
+    
+    func loadMoreIssues(containerWidth: CGFloat, completion: @escaping () -> ()) {
+        guard repo.hasIssuesEnabled else { return }
```

Comment:
I have seen these a couple of times in this PR (a `return` when the method accepts a completion handler). As a developer interacting with this API I would always expect my completion block to be called? 🤔  Is there something I am missing here that makes this intentional?